File: backend/src/main/java/com/ll/nbe342team8/domain/qna/question/dto/QuestionListDto.java
package com.ll.nbe342team8.domain.qna.question.dto;

import com.ll.nbe342team8.domain.qna.answer.dto.AnswerDto;
import com.ll.nbe342team8.domain.qna.question.entity.Question;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

//질문 목록 조회시 사용
public record QuestionListDto(

        @NotNull Long id,
        LocalDateTime createDate,
        LocalDateTime modifyDate,
        String title,
        String content,
        Boolean isAnswer


) {
    public QuestionListDto(Question question) {

        this(
                question.getId(),
                question.getCreateDate(),
                question.getModifyDate(),
                question.getTitle(),
                question.getContent(),
                question.getIsAnswer()
        );
    }

    public QuestionListDto(QuestionListDtoProjection questionListDtoProjection) {

        this(
                questionListDtoProjection.getId(),
                questionListDtoProjection.getCreateDate(),
                questionListDtoProjection.getModifyDate(),
                questionListDtoProjection.getTitle(),
                questionListDtoProjection.getContent(),
                questionListDtoProjection.getIsAnswer()
        );
    }
}

File: backend/src/main/java/com/ll/nbe342team8/domain/qna/question/dto/QuestionListDto.kt
package com.ll.nbe342team8.domain.qna.question.dto

import com.ll.nbe342team8.domain.qna.question.entity.Question
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime

//질문 목록 조회시 사용
data class QuestionListDto(
    val id: Long,
    val createDate: LocalDateTime,
    val title: String,
    val content: String,
    val isAnswer: Boolean


) {
    constructor(question: Question) : this(
        question.id,
        question.createDate,
        question.title,
        question.content,
        question.isAnswer
    )

    constructor(questionListDtoProjection: QuestionListDtoProjection) : this(
        questionListDtoProjection.getId(),
        questionListDtoProjection.getCreateDate(),
        questionListDtoProjection.getTitle(),
        questionListDtoProjection.getContent(),
        questionListDtoProjection.getIsAnswer()
    )
}